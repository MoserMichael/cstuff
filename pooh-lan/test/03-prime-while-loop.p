

# get all prime numbers up to max
sub primes( max )

  mark = []
  ret = []

  n = 2 
  while n < max
    m = 2 * n
    loop 
        if m > max
            break;
        end
        mark[ m ] = 1
        m = m + n
    end
    n = n + 1
  end

  n = 2
  while n  < max
    if mark[ n ] != 1
       push( ~array ret ~top n )
    end
    n = n + 1
  end

  return ret
end


# get prime numbers up to 100
pr = primes( ~max 100 )

# print them out.
i = 1
while i <= size( ~arg pr)
# println( ~msg 'prime number [ pr[i] ]' )
  println( ~msg 'prime number ' ..  pr[i]  )

  i = i + 1
end
