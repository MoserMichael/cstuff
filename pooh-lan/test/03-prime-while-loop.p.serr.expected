035|... = primes( ~max 100)...
006| mark = [ ] 
007| ret = [ ] 
009| n = 2
010| while (n:2 < max:100):true
011|  m = (2 * n:2):4
012|  loop
013|   if (m:4 > max:100):false
014|   end # if
Segmentation fault (core dumped)
